addons/udes_stock/models/stock_move_line.py

```diff
@@ -366,10 +360,152 @@ def prepare(
+                            "product_uom_qty": qty_done,
+                            "qty_done": qty_done,
+                            "lot_name": lot_id.name,
+                            "lot_id": lot_id.id,
+                        }
+                        prod_dict.update(vals)
+                        res[ml] = prod_dict
+                        quantity_swapping -= qty_done
+                else:
+                    qty_done = quantity_swapping
+                    # Update the first lot with all scanned in lot information
+                    ml, lot_id = next(mls_need_changing), next(lot_ids_swapped_in)
+                    reserved_quantity = ml.product_uom_qty
+                    prod_dict = {
+                        "product_uom_qty": qty_done,
+                        "qty_done": qty_done,
+                        "lot_name": lot_id.name,
+                        "lot_id": lot_id.id,
+                    }
+                    prod_dict.update(vals)
+                    res[ml] = prod_dict
+                    quantity_swapping -= reserved_quantity
+                    # If the lot quantity covers multiple movelines then delete these movelines
+                    # Unless it has partially picked some moveline then update the reserved qty on that moveline
+                    while quantity_swapping > 0:
+                        ml = next(mls_need_changing)
+                        reserved_quantity = ml.product_uom_qty
+                        quantity_swapping -= reserved_quantity
+                        if quantity_swapping >= 0:
+                            ml.unlink()
+                        else:
+                            prod_dict = {"product_uom_qty": abs(quantity), "qty_done": 0}
+                            res[ml] = prod_dict
+                
+                # Remove swapped in lot from old move line and replace it with swapped out lot
+                # Can zip as lot_ids_swapped_out will never have fewer item then swapped_lot_on_assigned_mls
+                for old_lot, mls_to_change in zip(lot_ids_swapped_out, swapped_lot_on_assigned_mls):
+                    # If we partially pick a lot that is reserved on another picking then we need to update the 
+                    # mls on that picking to have a new reserved quantity (old quantity - quantity swapped out) and create another moveline with
+                    # for the lot that was swapped out.
+                    if not is_serial and mls_to_change.product_uom_qty > amount_to_be_swapped:
+                        prod_dict = {"lot_name": mls_to_change.lot_id.name, "lot_id": mls_to_change.lot_id.id, "product_uom_qty": mls_to_change.product_uom_qty - amount_to_be_swapped, "qty_done": 0.0}
+                        mls_to_change.write({"product_uom_qty": mls_to_change.product_uom_qty - amount_to_be_swapped})
+                        res[mls_to_change] = prod_dict
+                        new_mls = mls_to_change.copy()
```

Comment:
Are we not using `_split()` because we don't want to `bypass_reservation_update` ?